### PR TITLE
feat: add support for assert in brillig

### PR DESCRIPTION
**Pull request by guipublic**
# Description

## Problem\*

assert keyword is not handled in brillig

Resolves 

## Summary\*

When an assert fail, the function fails execution.

This PR sets out to support the assert instruction in brillig.

### Example

This code:

```
x =false;
assert(x);
```
Before: 
was giving ICE: Instruction not supported 

After:
Error: could not satisfy all constraints

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: crates/noirc_evaluator/src/brillig/artifact.rs [hunk 1]
```diff
@@ -10,6 +10,10 @@ pub(crate) type JumpLabel = usize;
 /// particular basic block starts.
 pub(crate) type BlockLabel = usize;
 
+/// Number of bytecodes in the prefix
+/// The prefix code is added at the start of every brillig function
+const PREFIX_LEN: usize = 3;
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/brillig/artifact.rs:15]: Why is this needed?

Comment (guipublic) [reply crates/noirc_evaluator/src/brillig/artifact.rs:15]: This is needed for failing the assert.
Instead of doing:
(0): (...)
(x): if assert_is_true, jump x+2
(x+1): trap
(x+2):...

we do:
(0): jump (3)
(1): trap
(2):stop
(3): (...)
(x): if assert_is_false, jump 1
(x+1): ....

We save one opcode for each assert.

Comment (kevaundray) [reply crates/noirc_evaluator/src/brillig/artifact.rs:15]: We can apply optimizations like this after the fact, if we decide to add further brillig passes. Lets get assert to work and we can open an issue about optimizing the bytecode

Comment (ludamad) [reply crates/noirc_evaluator/src/brillig/artifact.rs:15]: I guess it's a question of what proper means, this definitely does not seem like the only possible jump location merging optimization

Comment (kevaundray) [reply crates/noirc_evaluator/src/brillig/artifact.rs:15]: To clarify, after the fact, was saying that we can apply optimizations to Brillig once we are done implementing. Whether we do it in-place or as a separate pass can be discussed when we are optimizing the bytecode.

The current implementation of Brillig generation should contain only the necessary items for generating Brillig

**Review by kevaundray — APPROVED** [review]
